### PR TITLE
Fix: change 0x data to nil of method id

### PR DESCRIPTION
**Pull request by Naupio**
link https://github.com/Magickbase/godwoken_explorer/issues/894

### CODE REVIEW

**Comment by Keith-CY** [comment]
More test cases are desirable

1. `input` has a valid `method id` so a hex string of 8 bytes should be returned
2. `input` is not long enough to represent a `method id` so `null` is expected
3. transaction is not a contract call so `null` is expected

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex [hunk 1]
```diff
@@ -37,18 +37,29 @@ defmodule GodwokenExplorer.Graphql.Resolvers.Transaction do
     from(t in Transaction, where: ^conditions)
   end
 
-  def method_id(%Transaction{hash: hash}, _args, _resolution) do
+  def method_id(%Transaction{hash: hash, to_account_id: to_account_id}, _args, _resolution) do
     p =
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/graphql/resolovers/transaction.ex:41]: 这种地方直接Repo.get_by 就好了，为何还要用where

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex [hunk 2]
```diff
@@ -37,18 +37,29 @@ defmodule GodwokenExplorer.Graphql.Resolvers.Transaction do
...
       |> where([p], p.tx_hash == ^hash)
       |> Repo.one()
 
-    if p do
-      input = to_string(p.input)
-      mid = String.slice(input, 0, 10)
-      Data.cast(mid)
-    else
-      {:ok, nil}
+    case Repo.get(Account, to_account_id) do
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/graphql/resolovers/transaction.ex:46]: 这里改用  with condition，因为其余三种情况都是{:ok, nil}

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_explorer/graphql/resolovers/transaction.ex]: fixed https://github.com/Magickbase/godwoken_explorer/pull/962/commits/5557f510556c3718489f71dcfd2650f5564e908f

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex (hunk 2 above)
Comment (Naupio) [reply lib/godwoken_explorer/graphql/resolovers/transaction.ex]: fixed https://github.com/Magickbase/godwoken_explorer/pull/962/commits/5557f510556c3718489f71dcfd2650f5564e908f

File: lib/godwoken_explorer/graphql/resolovers/transaction.ex [hunk 3]
```diff
@@ -37,26 +37,21 @@ defmodule GodwokenExplorer.Graphql.Resolvers.Transaction do
...
-      |> Repo.one()
-
-    if p do
-      input = to_string(p.input)
-      mid = String.slice(input, 0, 10)
+  def method_id(%Transaction{hash: hash, to_account_id: to_account_id}, _args, _resolution) do
+    with %Account{type: :polyjuice_contract} <- Repo.get(Account, to_account_id),
+         p when not is_nil(p) <- Repo.get_by(Polyjuice, tx_hash: hash),
+         input when is_bitstring(input) <- p.input |> to_string(),
+         mid <- String.slice(input, 0, 10),
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer/graphql/resolovers/transaction.ex:44]: `length when length >= 10 <- input |> String.slice(0, 10) |> String.length()`

**Review by zmcNotafraid — APPROVED** [review]

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**